src/CellData/CellFields.jl

```diff
@@ -204,14 +204,97 @@ DomainStyle(::Type{GenericCellField{DS}}) where DS = DS()
 
 (a::CellField)(x) = evaluate(a,x)
 
+struct CellFieldCache
+  kdtree::KDTree{SVector{D,T},Euclidean,T} where {D,T}
+end
+
+function return_cache(f::CellField,x::Point)
+  # Create k-d tree for mesh
+  trian = get_triangulation(f)
+  node_coordinates = get_node_coordinates(trian)
+  @assert !isempty(node_coordinates)
+  sample_coord = node_coordinates[1]
+  D = length(sample_coord)
+  T = eltype(sample_coord)
+  @assert D > 0
+  kdtree = KDTree(map(nc -> SVector(nc.data), node_coordinates))
+  kdtree::KDTree{SVector{D,T},Euclidean,T}
+  return CellFieldCache(kdtree)
+end
+
 function evaluate!(cache,f::CellField,x::Point)
-  @notimplemented """\n
-  Evaluation of a CellField at a given Point is not implemented yet.
+  cache::CellFieldCache
+
```

Comment:
We could probably encapsulate this code in a function and do dispatching with respect to the triangulation type. The code below is for an unstructured (simplicial) mesh. We could also consider the case for a structured Cartesian mesh (which is much easier, no tree involved).